spec/duration.html

```diff
@@ -2017,28 +2017,29 @@ <h1>
           1. If _unit_ is not *"week"*, or _largestUnit_ is *"week"*, then
             1. If _unit_ is *"year"*, then
               1. Let _years_ be _duration_.[[Years]] + _sign_.
-              1. Let _endDuration_ be ? CreateNormalizedDurationRecord(_years_, 0, 0, 0, ZeroTimeDuration()).
+              1. Let _endDuration_ be ? CreateDateDurationRecord(_years_, 0, 0, 0).
             1. Else if _unit_ is *"month"*, then
               1. Let _months_ be _duration_.[[Months]] + _sign_.
-              1. Let _endDuration_ be ? CreateNormalizedDurationRecord(_duration_.[[Years]], _months_, 0, 0, ZeroTimeDuration()).
+              1. Let _endDuration_ be ? CreateDateDurationRecord(_duration_.[[Years]], _months_, 0, 0).
             1. Else if _unit_ is *"week"*, then
               1. Let _weeks_ be _duration_.[[Weeks]] + _sign_.
-              1. Let _endDuration_ be ? CreateNormalizedDurationRecord(_duration_.[[Years]], _duration_.[[Months]], _weeks_, 0, ZeroTimeDuration()).
+              1. Let _endDuration_ be ? CreateDateDurationRecord(_duration_.[[Years]], _duration_.[[Months]], _weeks_, 0).
             1. Else,
               1. Assert: _unit_ is *"day"*.
               1. Let _days_ be _duration_.[[Days]] + _sign_.
-              1. Let _endDuration_ be ? CreateNormalizedDurationRecord(_duration_.[[Years]], _duration_.[[Months]], _duration_.[[Weeks]], _days_, ZeroTimeDuration()).
-            1. Let _end_ be ? AddDateTime(_dateTime_.[[Year]], _dateTime_.[[Month]], _dateTime_.[[Day]], _dateTime_.[[Hour]], _dateTime_.[[Minute]], _dateTime_.[[Second]], _dateTime_.[[Millisecond]], _dateTime_.[[Microsecond]], _dateTime_.[[Nanosecond]], _calendarRec_, _endDuration_.[[Years]], _endDuration_.[[Months]], _endDuration_.[[Weeks]], _endDuration_.[[Days]], _endDuration_.[[NormalizedTime]], *undefined*).
+              1. Let _endDuration_ be ? CreateDateDurationRecord(_duration_.[[Years]], _duration_.[[Months]], _duration_.[[Weeks]], _days_).
+            1. Let _date_ be ! CreateTemporalDate(_dateTime_.[[Year]], _dateTime_.[[Month]], _dateTime_.[[Day]], _calendarRec_.[[Receiver]]).
+            1. Let _end_ be ? AddDate(_calendarRec_, _date_, ! CreateTemporalDuration(_endDuration_.[[Years]], _endDuration_.[[Months]], _endDuration_.[[Weeks]], _endDuration_.[[Days]], 0, 0, 0, 0, 0, 0)).
             1. If _timeZoneRec_ is ~unset~, then
-              1. Let _endEpochNs_ be GetUTCEpochNanoseconds(_end_.[[Year]], _end_.[[Month]], _end_.[[Day]], _end_.[[Hour]], _end_.[[Minute]], _end_.[[Second]], _end_.[[Millisecond]], _end_.[[Microsecond]], _end_.[[Nanosecond]]).
+              1. Let _endEpochNs_ be GetUTCEpochNanoseconds(_end_.[[Year]], _end_.[[Month]], _end_.[[Day]], _dateTime_.[[Hour]], _dateTime_.[[Minute]], _dateTime_.[[Second]], _dateTime_.[[Millisecond]], _dateTime_.[[Microsecond]], _dateTime_.[[Nanosecond]]).
```

Comment:
Likewise, `_end_.[[ISOYear]], _end_.[[ISOMonth]], _end_.[[ISODay]]` here and below.

Reply:
Also fixed.